testsuite/gateway/gateway_api/gateway.py

```diff
@@ -106,6 +106,23 @@ def get_tls_cert(self, hostname):
         )
         return tls_cert
 
+    def get_tls_secret(self, hostname):
```

Comment:
Please make method `get_tls_cert()` call this method to get its secret to prevent duplicated code